### PR TITLE
Fb notification other answers

### PR DESCRIPTION
**Pull request by ElsaWeb**
Users will now receive notifications when someone else answers the same offer as them, so that they can see it and potentially improve their proposition.
This pr solves issue #185.

### CODE REVIEW

File: app/src/main/java/ch/epfl/sweng/swenggolf/offer/ShowOfferActivity.java [hunk 1]
```diff
@@ -408,9 +409,21 @@ public void postAnswer(View view) {
                         new Notification(NotificationType.ANSWER_POSTED,
                                 Config.getUser(), offer));
             }
+            sendNotificationToPreviousAnswerers(answers.getAnswerList());
             listAnswerAdapter.notifyDataSetChanged();
         }
+    }
 
+    private void sendNotificationToPreviousAnswerers(List<Answer> answerList) {
```
Comment (SFulpius) [inline app/src/main/java/ch/epfl/sweng/swenggolf/offer/ShowOfferActivity.java]: If the creator of the offer leave an answer, he will receive a notification. But he already receives a notification when someone leave an answer. Thus he will receive two notifications when someone leave an answer.

File: app/src/main/java/ch/epfl/sweng/swenggolf/offer/ShowOfferActivity.java [hunk 2]
```diff
@@ -408,9 +409,21 @@ public void postAnswer(View view) {
...
+        List<String> participants = new ArrayList<>();
```
Comment (SFulpius) [inline app/src/main/java/ch/epfl/sweng/swenggolf/offer/ShowOfferActivity.java]: In this case, you're only adding and testing if the collection contains data. A set would be more efficient for this kind of usage.

File: app/src/main/java/ch/epfl/sweng/swenggolf/offer/ShowOfferActivity.java (hunk 1 above)
Comment (ElsaWeb) [reply app/src/main/java/ch/epfl/sweng/swenggolf/offer/ShowOfferActivity.java]: Oh yes you're right! I did not think about that.

File: app/src/main/java/ch/epfl/sweng/swenggolf/offer/ShowOfferActivity.java (hunk 2 above)
Comment (ElsaWeb) [reply app/src/main/java/ch/epfl/sweng/swenggolf/offer/ShowOfferActivity.java]: This is absolutely correct thanks.

**Pull request merged**